src/main/java/thePackmaster/actions/graveyardpack/PlayFromExhaustAction.java

```diff
@@ -0,0 +1,156 @@
+package thePackmaster.actions.graveyardpack;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.MakeTempCardInDiscardAction;
+import com.megacrit.cardcrawl.actions.common.MakeTempCardInDrawPileAction;
+import com.megacrit.cardcrawl.actions.utility.NewQueueCardAction;
+import com.megacrit.cardcrawl.actions.utility.QueueCardAction;
+import com.megacrit.cardcrawl.actions.utility.ShowCardAndPoofAction;
+import com.megacrit.cardcrawl.actions.utility.WaitAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.CardGroup;
+import com.megacrit.cardcrawl.cards.CardQueueItem;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.localization.UIStrings;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import com.megacrit.cardcrawl.vfx.cardManip.ShowCardBrieflyEffect;
+
+import thePackmaster.actions.graveyardpack.CleanUpCardAction;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+
+import static thePackmaster.SpireAnniversary5Mod.makeID;
+
+
+
+public class PlayFromExhaustAction extends AbstractGameAction
+{
+	private AbstractPlayer p;
+	private ArrayList<AbstractCard> classless;
+	private static final UIStrings uiStrings = CardCrawlGame.languagePack.getUIString(makeID("PlayFromExhaustAction"));
+	public static final String[] TEXT = uiStrings.TEXT;
+	
+	public PlayFromExhaustAction() {
+		this.classless = new ArrayList();
+		this.p = AbstractDungeon.player;
+		setValues(this.p, AbstractDungeon.player, this.amount);
+		this.actionType = AbstractGameAction.ActionType.CARD_MANIPULATION;
+		this.duration = Settings.ACTION_DUR_FAST;
+	}
+	  
+	public void update() {
+    	if (this.duration == Settings.ACTION_DUR_FAST) {
+  	      if (this.p.exhaustPile.isEmpty()) {
+  	        this.isDone = true;
+  	        return;
+  	      }
+  	      
+  	    	for (Iterator<AbstractCard> ex = this.p.exhaustPile.group.iterator(); ex.hasNext(); ) {
+  		        AbstractCard derp = (AbstractCard)ex.next();
+  		        if (derp.color.equals(AbstractCard.CardColor.COLORLESS) || derp.color.equals(AbstractCard.CardColor.CURSE)) {
+  		          this.classless.add(derp);
+  		          ex.remove();
+  		        } 
+  		    }
+  	    	
+  	      if (this.p.exhaustPile.isEmpty()) {
+  	    	  this.p.exhaustPile.group.addAll(classless);
```

Comment:
If the pile is empty, add everything from it? I think there's a problem here.

Reply:
Not a bug. This adds things back into the exhaust pile - it seems odd, but I think it's the straightforward way to use the original game's method for picking cards from the exhaust without letting the player pick colorless cards.

Reply:
Oh I see, I got confused by the different piles